edgemesh/pkg/resolver/http_resolver.go

```diff
@@ -0,0 +1,92 @@
+package resolver
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"net/http"
+
+	"github.com/go-chassis/go-chassis/core/handler"
+	"github.com/go-chassis/go-chassis/core/invocation"
+	"github.com/kubeedge/beehive/pkg/common/log"
+	"github.com/kubeedge/kubeedge/edgemesh/pkg/common"
+)
+
+type HTTPResolver struct{}
+
+func (resolver *HTTPResolver) Resolve(data chan []byte, stop chan interface{}, invCallback func(string, invocation.Invocation, []string, bool)) (invocation.Invocation, bool) {
+	content := ""
+	protocol := ""
+	for {
+		select {
+		case d := <-data:
+			strData := string(d[:])
+			if protocol == "" {
+				if common.IsHTTPRequest(strData) {
+					protocol = "http"
+				} else {
+					return invocation.Invocation{}, false
+				}
+			}
+			content += strData
+			req, err := http.ReadRequest(bufio.NewReader(bytes.NewReader([]byte(content))))
+			if err == nil {
+				content = ""
+				req.RequestURI = ""
+				i := invocation.New(context.Background())
```

Comment:
This function is too large, create a function called NewInvocation that returns an invocation please